### PR TITLE
tools/ports: supports embuilder --pic build harfbuzz

### PR DESCRIPTION
**Pull request by pmp-p**
quick fix for embuilder --pic build harfbuzz could fail without -fpic in compiler flags
could close https://github.com/emscripten-core/emsdk/issues/479

### CODE REVIEW

File: tools/ports/harfbuzz.py [hunk 1]
```diff
@@ -44,8 +44,16 @@ def create():
       '-DHB_HAVE_FREETYPE=ON'
     ]
 
+    extra_cxx_flags = []
+
+    if TAG.find('pic')>=0:
```
Comment (sbc100) [inline tools/ports/harfbuzz.py:49]: Should this be `settings.RELOCATABLE`?

Comment (pmp-p) [reply tools/ports/harfbuzz.py]: looks like it, thanks

**Comment by pmp-p** [comment]
@kripken , actually the pic builds are in totally different cache folder ( /wasm-pic/ ) based on TAG so there's really no need for extra suffix.

**Comment by kripken** [comment]
Oh right, thanks @pmp-p - I forgot how differently atomics and relocatable are handled.

**Comment by pmp-p** [comment]
sorry, patch was incomplete, some object files were not pic compiled

**Comment by kripken** [comment]
>  ( does cache survive emsdk update ? )

It shouldn't, I think (assuming the update bumped the version).

**Comment by sbc100** [comment]
cache should be completely replaced by installation/unzip action since it lives inside the `upstream/emscripten/cache`, which itself lives inside `upstream` which should be 100% replaced by any installation.

**Pull request merged**